cordyceps/src/sorted_list.rs

```diff
@@ -0,0 +1,459 @@
+    ///
+    /// * Consider using [`SortedList::new_min()`] if you want **smallest** items sorted first.
+    /// * Consider using [`SortedList::new_custom()`] if you want to provide your own sorting
+    ///   implementation.
+    ///
+    /// If two items are considered of equal value, new values will be placed AFTER
+    /// old values.
+    pub const fn new_max() -> Self {
+        Self::new_custom(invert_sort::<T>)
+    }
+
+    /// Create a new sorted list, consuming the stack, sorted GREATEST FIRST
+    pub fn from_stack_max(stack: Stack<T>) -> Self {
+        Self::from_stack_custom(stack, invert_sort::<T>)
+    }
+}
+
+impl<T: Linked<Links<T>>> SortedList<T> {
+    /// Create a new (empty) sorted list with the given ordering function
+    ///
+    /// If your type T implements [`Ord`]:
+    ///
+    /// * Consider using [`SortedList::new_min()`] if you want **smallest** items sorted first.
+    /// * Consider using [`SortedList::new_max()`] if you want **largest** items sorted first.
+    ///
+    /// If `T` contained an `i32`, and you wanted the SMALLEST items at the
+    /// front, then you could provide a function something like:
+    ///
+    /// ```rust
+    /// let f: fn(&i32, &i32) -> core::cmp::Ordering = |lhs, rhs| {
+    ///     lhs.cmp(rhs)
+    /// };
+    /// ```
+    ///
+    /// SortedList takes a function (and not just [Ord]) so you can use it on types
+    /// that don't have a general way of ordering them, allowing you to select a
+    /// specific metric within the sorting function.
+    ///
+    /// If two items are considered of equal value, new values will be placed AFTER
+    /// old values.
+    pub const fn new_custom(f: fn(&T, &T) -> core::cmp::Ordering) -> Self {
+        Self {
+            func: f,
+            head: None,
+        }
+    }
+
+    /// Create a new sorted list, consuming the stack, using the provided ordering function
+    pub fn from_stack_custom(stack: Stack<T>, f: fn(&T, &T) -> core::cmp::Ordering) -> Self {
+        let mut slist = Self::new_custom(f);
+        slist.extend(stack);
+        slist
+    }
+
+    /// Pop the front-most item from the list, returning it by ownership (if it exists)
+    ///
+    /// This is an `O(1)` operation.
+    #[must_use]
+    pub fn pop_front(&mut self) -> Option<T::Handle> {
+        test_trace!(?self.head, "SortedList::pop_front");
+        let head = self.head.take()?;
+        unsafe {
+            // Safety: we have exclusive ownership over this chunk of stack.
+
+            // advance the head link to the next node after the current one (if
+            // there is one).
+            self.head = T::links(head).as_mut().next.with_mut(|next| (*next).take());
+
+            test_trace!(?self.head, "SortedList::pop -> popped");
+
+            // return the current node
+            Some(T::from_ptr(head))
+        }
+    }
+
+    /// Insert a single item into the list, in it's sorted order position
+    ///
```

Comment:
it would be nice if we mentioned the behavior where two items are equal in this doc comment, too.